source/adaptive_load/adaptive_load_controller_impl.cc

```diff
@@ -84,6 +85,22 @@ StepControllerPtr LoadStepControllerPluginFromSpec(const AdaptiveLoadSessionSpec
   return std::move(step_controller_or.value());
 }
 
+/**
+ * Converts the duration proto to a number of milliseconds.
+ *
+ * @param duration the duration to convert.
+ *
+ * @return the number of milliseconds or an error if the conversion failed.
+ */
+absl::StatusOr<uint64_t>
+ProtobufDurationToMilliseconds(const Envoy::ProtobufWkt::Duration& duration) noexcept {
+  try {
+    return Envoy::DurationUtil::durationToMilliseconds(duration);
```

Comment:
Elsewhere in this file we are using a conversion function that doesn't throw an exception:
https://github.com/envoyproxy/nighthawk/blob/cf13dd102b8f6b28bec864b8785a62c972168e60/source/adaptive_load/adaptive_load_controller_impl.cc#L175

It may or may not make things simpler here.

Reply:
Done in 4678ada.